app/Export/GetImages.hs

```diff
@@ -1,25 +1,98 @@
+-- [["8:00","","","","",""],["9:00","","","","",""],["10:00","CSC108 (L)","","CSC108 (L)","","CSC108 (L)"],["11:00","","","","",""],["12:00","","","","",""],["1:00","","","","",""],["2:00","STA355 (L)","","STA355 (L)","",""],["3:00","STA355 (L)","","","",""],["4:00","","","","",""],["5:00","","","","",""],["6:00","","","","",""],["7:00","","","","",""],["8:00","","","","",""]]
+getActiveTimetable :: Request -> IO (String, String, String, String)
+getActiveTimetable req = do
+    let cookies = M.fromList $ rqCookies req  --  Map String Cookie
+        coursecookie = maybe "" cookieValue $ M.lookup "selected-lectures" cookies
+        (lectures, tutorials) = parseCourseCookie coursecookie
+    -- liftIO $ print coursecookie
+    -- liftIO $ print lectures
+    -- liftIO $ print tutorials
+    lecTimes <- mapM getLectureTime lectures  -- [[string]] -> IO ([Maybe [Time]])
+    tutTimes <- mapM getTutorialTime tutorials
+    let lecture_times = zip lectures lecTimes
+        tutorial_times = zip tutorials tutTimes
+        all_times = lecture_times ++ tutorial_times
+        (fall_times, spring_times) = partition isFall all_times
+        fall_schedule = replicate 13 $ replicate 5 ""
+        spring_schedule = replicate 13 $ replicate 5 ""
+        fall_schedule' = foldl (\acc x -> addCourseToSchedule x acc) fall_schedule fall_times
+        spring_schedule' = foldl (\acc x -> addCourseToSchedule x acc) spring_schedule spring_times
+    fallrand <- randomName
+    springrand <- randomName
+    let fallsvgFilename = fallrand ++ ".svg"
+        fallimageFilename = fallrand ++ ".png"
+        springsvgFilename = springrand ++ ".svg"
+        springimageFilename = springrand ++ ".png"
+    renderTableHelper fallsvgFilename (zipWith (:) times fall_schedule') "Fall"
+    renderTableHelper springsvgFilename (zipWith (:) times spring_schedule') "Spring"
+    createImageFile fallsvgFilename fallimageFilename
+    createImageFile springsvgFilename springimageFilename
+    return (fallsvgFilename, fallimageFilename, springsvgFilename, springimageFilename)
+    where isFall (c, t) = last c == "F"
+    
+
+
+-- "CSC148H1-L5101-S_CSC148H1-T0501-S_STA355H1-L0101-F_CSC108H1-L0102-F"
+-- ([["CSC148H1","L5101","S"],["STA355H1","L0101","F"],["CSC108H1","L0102","F"]],  [["CSC148H1","T0501","S"]])
```

Comment:
I recommend replacing "L" with "LEC" and "T" with "TUT" (and "P" with "PRA") here, to clean up the database lookup after.